src/Notifier.js

```diff
@@ -0,0 +1,327 @@
+        body: {
+          body: `${data.discount}% off • ${data.total - data.sold} remaining\nEnds in ${data.eta}`,
+          icon: wfcdLogoUrl,
+        },
+        type: 'darvo',
+      };
+    case 'enemies':
+      return {
+        head: `${data.agentType} discovered!`,
+        body: {
+          body: `${(data.healthPercent * 100).toFixed(2)}% Health Remaining • Discovered at ${data.lastDiscoveredAt}`,
+          icon: wfcdLogoUrl,
+        },
+        type: 'enemies',
+      };
+    case 'sortie':
+      return {
+        head: `Sortie: ${data.boss}`,
+        body: {
+          body: `${data.variants.map((variant) => `${variant.missionType} • ${variant.node} • ${variant.modifier}`).join('\n')}\n${data.eta}`,
+          icon: wfcdLogoUrl,
+        },
+      };
+    case 'fissure':
+      return {
+        head: 'New Fissure Detected',
+        body: {
+          body: `${data.tier} ${data.missionType} • ${data.node}`,
+          icon: wfcdLogoUrl,
+        },
+        type: 'fissure',
+      };
+    case 'news':
+      return {
+        head: 'Warframe Hub',
+        body: {
+          body: data.message,
+          icon: wfcdLogoUrl,
+        },
+        link: data.link,
+        type: 'news',
+      };
+    case 'invasions':
+      var rewards = `${data.attackerReward.asString.length ? `${data.attackerReward.asString} vs ` : ''}${data.defenderReward.asString}`;
+
+      return {
+        head: rewards,
+        body: {
+          body: `${data.desc} • ${data.node}\n${data.eta.replace('-Infinityd', '??').replace('Infinityd', '??')} Remaining`,
+          icon: wfcdLogoUrl,
+        },
+        sound: 'drum',
+        type: 'invasion',
+      };
+    case 'vallis.warm':
+      return {
+        head: 'Vallis - Get your Warm Fishing on!',
+        body: {
+          body: data.shortString,
+          icon: wfcdLogoUrl,
+        },
+        type: 'warm',
+      };
+    case 'vallis.cold':
+      return {
+        head: 'Vallis - Back to the Cold!',
+        body: {
+          body: data.shortString,
+          icon: wfcdLogoUrl,
+        },
+        type: 'cold',
+      };
+    case 'nightwave':
+      return {
+        head: `Nightwave - ${data.title}`,
+        body: {
+          body: `${data.daily ? 'Daily: ' : (data.isElite ? 'Weekly Elite: ' : 'Weekly: ')}${data.desc}`,
+          icon: wfcdLogoUrl,
+        },
+      };
+    default:
+      return defaultNotificationBody;
+  }
+};
+
+class Notifier {
+  constructor (store) {
+    this.store = store;
+    this.notifier = Vue.notification;
+  }
+
+  isNotifiable(id, event, items) {
+    const currentNotifieds = this.store.getters.notifiedIds;
+    const eventGood = !currentNotifieds.includes(id) && this.trackedEvents.includes(event);
+    const includesItems = items && items.length ? this.trackedRewards.some((r) => items.indexOf(r) >= 0) : true;
+    return eventGood && includesItems;
+  }
+
+  async notify (notifications) {
+    const usePush = true || !document.hasFocus();
```

Comment:
This looks to be always true.

Reply:
yeah, i want to add an option for native in-app notifications, but that's down the line. i'll add a note and put in an issue